SqlInstall.ps1

```diff
@@ -0,0 +1,556 @@
+                Write-Host "BackupVolume remains on " $BackupVolume
+            }
+        }
+        Default {Write-Host "Drives agreed, continuing"; }
+    }
+
+    $CheckLastTwoChar = ":\"
+    $CheckLastChar = "\"
+
+    $SetupFile = Read-Host -Prompt 'Please enter the root location for Setup.exe'
+    IF ($SetupFile.Length -gt 1) {
+        $C2 = $SetupFile.Substring($SetupFile.Length - 2)
+        $C1 = $SetupFile.Substring($SetupFile.Length - 1)
+        If ($C2 -eq $CheckLastTwoChar) {
+            $debug = $SetupFile.Substring($SetupFile.Length - 2)
+            Write-Host $debug '/' $CheckLastTwoChar
+            $SetupFile = $SetupFile.Substring(0, $SetupFile.Length - 2)
+            Write-Host $SetupFile
+        }
+        elseif ($C1 -eq $CheckLastChar) {
+            $SetupFile = $SetupFile.Substring(0, $SetupFile.Length - 1)
+            Write-Host $SetupFile
+        }
+    }
+    IF ($SetupFile.Length -eq 1) {
+        $SetupFile = $SetupFile + ':\SQLEXPR_x64_ENU\SETUP.EXE'
+        Write-Host 'Setup will start from ' + $SetupFile
+    } 
+    else {
+        $SetupFile = $SetupFile + '\SQLEXPR_x64_ENU\SETUP.EXE'
+        Write-Host 'Setup will start from ' + $SetupFile
+    }
+
+
+    $ConfigFile = 'c:\temp\'
+
+    if ( -Not (Test-Path -Path $ConfigFile ) ) {
+        New-Item -ItemType directory -Path $ConfigFile
+    }
+
+    Out-File -FilePath C:\Temp\ConfigurationFile2.ini -InputObject $startScript
+
+    $FileLocation2 = $ConfigFile + 'ConfigurationFile2.ini'
+
+    (Get-Content -Path $FileLocation2).Replace('SQLBACKUPDIR="E:\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Backup"', 'SQLBACKUPDIR="' + $BackupVolume + ':\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Backup"') | Out-File $FileLocation2
+
+    (Get-Content -Path $FileLocation2).Replace('SQLUSERDBDIR="E:\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Data"', 'SQLUSERDBDIR="' + $DataVolume + ':\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Data"') | Out-File $FileLocation2
+
+    (Get-Content -Path $FileLocation2).Replace('SQLTEMPDBDIR="E:\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Data"', 'SQLTEMPDBDIR="' + $TempVolume + ':\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Data"') | Out-File $FileLocation2
+
+    (Get-Content -Path $FileLocation2).Replace('SQLUSERDBLOGDIR="E:\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Log"', 'SQLUSERDBLOGDIR="' + $LogVolume + ':\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Log"') | Out-File $FileLocation2
+
+    (Get-Content -Path $FileLocation2).Replace('SQLSYSADMINACCOUNTS="WIN-NAJQHOBU8QD\Administrator"', 'SQLSYSADMINACCOUNTS="' + $env:COMPUTERNAME + '\Administrator"')| Out-File $FileLocation2
+
+    #$SetupFile = 'C:\Users\Administrator\Downloads\SQLEXPR_x64_ENU\Setup.exe'
+    #$ConfigFile = 'C:\temp\ConfigurationFile2.ini'
+
+    $SAPassW = '[InsertPasswordHere]'
```

Comment:
We should try to avoid plain text when possible. Use a [PsCredential] (Get-Credential) and then `$credential.GetNetworkCredential().password`